### PR TITLE
[grafana] Make deployable in Azure with Makefile

### PR DESCRIPTION
**Pull request by daniel-goldstein**
The Makefile was hardcoded for GCP. Grafana is still currently able to be deployed through CI, this just changes the Makefile so it's not hardcoded to `gcp` and moves the gcp project from a Makefile-set value to one set through kubernetes so the Makefile is more cloud agnostic (doesn't reference a gcp project).

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -1923,6 +1923,7 @@ steps:
     config: grafana/deployment.yaml
     scopes:
       - deploy
+      - test
```
Comment (jigold) [inline build.yaml:1926]: Was there a reason for not including this scope initially? Maybe something about needing disks?

File: grafana/deployment.yaml [hunk 2]
```diff
@@ -138,7 +145,7 @@ data:
         isDefault: true
         jsonData:
           authenticationType: gce
-          defaultProject: {{ global.gcp_project }}
+          defaultProject: $GCP_PROJECT
```
Comment (jigold) [inline grafana/deployment.yaml:148]: I'm trying to figure out why this change would work. Can you explain it to me?

**Review by jigold — CHANGES_REQUESTED** [review]

File: build.yaml (hunk 1 above)
Comment (daniel-goldstein) [reply build.yaml:1926]: My guess is that this was removed along with the `test` scope for prometheus, which was removed iirc because of it needs an entire k8s node for itself. But based off this PR run it looks like grafana doesn't need prometheus to exist for it to successfully start. We might even be able to deploy prometheus in tests too and just change the resource requirements for test deploys, since it'll be starting from scratch and not have to load a ton of data in PR runs. I can try that out too if you want.

File: grafana/deployment.yaml (hunk 2 above)
Comment (daniel-goldstein) [reply grafana/deployment.yaml:148]: I just learned that grafana allows [environment variables](https://grafana.com/docs/grafana/latest/administration/provisioning/#using-environment-variables) in their datasource templates and it will fill them in when grafana loads the yaml. So as long as `GCP_PROJECT` is set as an environment variable in the grafana container it should just work.

File: build.yaml (hunk 1 above)
Comment (daniel-goldstein) [reply build.yaml:1926]: hm going back in the git history it looks like this just never had the `test` scope

Comment (jigold) [reply build.yaml:1926]: I'm all for testing grafana in CI, but I do want to be mindful of resources. If you can give it sensible parameters in testing, then this seems fine to me. If we don't like the behavior, then we can revert.

Comment (daniel-goldstein) [reply build.yaml:1926]: It's requesting a modest amount of resources, and its disk is only 10Gi. I think it should have a negligible impact on our testing resources. Prometheus however gives me more pause, since it accumulates a lot of data quickly and I have yet to test how modest we can be within the timeline that the PR deployment would live. Maybe I'll save testing prometheus for the new year and just stick to grafana for now?

Comment (jigold) [reply build.yaml:1926]: that seems fine.

**Review by jigold — APPROVED** [review]
